pkg/volume/iscsi/iscsi.go

```diff
@@ -283,13 +306,74 @@ func portalMounter(portal string) string {
 	return portal
 }
 
-func getVolumeSource(spec *volume.Spec) (*v1.ISCSIVolumeSource, bool, error) {
+// get iSCSI volume info: readOnly and fstype
+func getISCSIVolumeInfo(spec *volume.Spec) (bool, string, error) {
 	if spec.Volume != nil && spec.Volume.ISCSI != nil {
-		return spec.Volume.ISCSI, spec.Volume.ISCSI.ReadOnly, nil
+		return spec.Volume.ISCSI.ReadOnly, spec.Volume.ISCSI.FSType, nil
 	} else if spec.PersistentVolume != nil &&
 		spec.PersistentVolume.Spec.ISCSI != nil {
-		return spec.PersistentVolume.Spec.ISCSI, spec.ReadOnly, nil
+		return spec.ReadOnly, spec.PersistentVolume.Spec.ISCSI.FSType, nil
```

Comment:
Why this refers spec.ReadOnly? Shouldn't we get spec.PersistentVolume.Spec.ISCSI.ReadOnly?

Reply:
it is probably not documented but PV spec's readOnly overrides volume source spec's.

Reply:
That should be documented.

Reply:
since the type is different now, we have a place to document that